### DISCUSSION TITLE
BMW location as address in Home assistant?

### DISCUSSION THREAD

**Discussion by richtj999** [bimmerconnected/bimmer_connected#620] [Q&A]
Hi,
This might be obvious but am having issues figuring it out.  In home assistant i have home as a location under "Zones" so if the car is home i know.  Is there a way of showing an address or map point with the BMW integration?  My son borrows my car periodically and i would like to know where it is within HA.

I am guessing if home is set and it knows if the car is home or away then the address is somewhere within HA.  I just dont know where to find this function.

Thanks for the help,
Rich

**Comment by rikroe** [comment]
Shouldn't your car be displayed on the integrated map dashboard?
Otherwise you could add a `map` card specifically for this.

**Reply by richtj999** [reply]
Thanks for getting back to me.  I am a little unclear - but should i have an option for a map view within the Home Assistant devices/integration/bmw area?

Alternatively - how do i add your integration to the map card within home assistant?

Thanks - this is very helpful!

**Reply by sincze** [reply]
![image](https://github.com/bimmerconnected/bimmer_connected/assets/5776333/5aa3cc49-93e2-4e15-a0b9-5849fe93069b)
Something like this.

**Reply by rikroe** [reply]
If you look a the default [map dashboard](https://www.home-assistant.io/dashboards/dashboards/#map-dashboard) or create you own dashboard with the map component, you will see an icon with your car.
 
![image](https://github.com/bimmerconnected/bimmer_connected/assets/42204099/f8c8a889-5fe7-4753-aae2-91d94f187dcb)

**Reply by richtj999** [reply]
Thank you - i did give that a try - i am guessing the default is a view live at the moment.  Is there a way to see the history of a vehicles location?

**Reply by sincze** [reply]
I store historic routes in 'Traccar' but I can try to add it to the recorder.yaml in HA for a similar result.

**Reply by rikroe** [reply]
First entry for googling "home assistant location history" ;)
https://community.home-assistant.io/t/how-to-configure-the-history-in-map/189206

So when adding a second dashboard with a map component, you can define the amount of hours it should show in the past.
